components/Python/PySpark/slack-events/main.py

```diff
@@ -0,0 +1,157 @@
+import os
+import json
+import argparse
+import traceback
+from datetime import datetime, timedelta
+from parallelm.mlops import mlops as pm
+from slackclient import SlackClient
+from parallelm.mlops.stats.opaque import Opaque
+from pyspark.sql import SparkSession
+from pyspark import SparkContext
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--token", help="slack token")
+    parser.add_argument("--channel-name", help="channel to write messages to")
+    parser.add_argument("--alert", help="show only alerts if set to True")
+    options = parser.parse_args()
+    return options
+
+def get_my_agents():
+    """
+    Return agent list of current ION node
+    :return: Agent list used by current ION node
+    """
+
+    # Getting the first agent of ion component "0"
+    agent_list = pm.get_agents(pm.get_current_node().name)
+    if len(agent_list) == 0:
+        print("Error - must have agents this ion component is running on")
+        raise Exception("Agent list is empty")
+    return agent_list
+
+def get_channel_id(sc, name):
+    #ret = sc.api_call("channels.list")
+    ret = sc.api_call('conversations.list', types='private_channel,public_channel')
+
+    chid = None
+    try:
+        for ch in ret['channels']:
+            if ch['name'] == 'testapp':
+                print(ch)
+                chid = ch['id']
+    except Exception:
+        print(traceback.format_exc())
+    return chid
+
+def main():
+    spark = SparkSession.builder.appName("Slack").getOrCreate()
```

Comment:
Move to python

Reply:
ill test it on python before updating here - until then ill leave it as pyspark

Reply:
switched to python (pr updated)